src/utils/connected-components.ts

```diff
@@ -0,0 +1,30 @@
+import { IGraph } from '..'
+
+/** Find the connected components of the clique graph.  Complexity is on the
+ * order O(n m) where n is the number of cliques and m is the number of separation
+ * sets.   This complexity could be reduced to n (log m) by using a more efficient
+ * means of finding
+ */
+export const getConnectedComponents = (junctionTree: IGraph) => {
+  const ids: string[] = junctionTree.getNodesId()
+  const visited: Set<string> = new Set<string>()
+  const ccs = []
+
+  const process = (id: string): string[] => {
+    const result = [id]
+    visited.add(id)
+    const neighbors = junctionTree.getNodeEdges(id).filter((z: string) => !visited.has(z))
+
+    for (const neighbor of neighbors) {
+      result.push(...process(neighbor))
+    }
+
+    return result
+  }
```

Comment:
Just another way to write it. 
```ts
  const process = (id: string): string[] => {
    const neighbors = junctionTree.getNodeEdges(id)
    const notVisitedNeighbors = neighbors.filter((z: string) => !visited.has(z))
    
    visited.add(id)
    
    return reduce(
      (result, neighbor) => {
        result.push(...process(neighbor))
        return result
      },
      [id],
      notVisitedNeighbors
    )
  }
```

**But you don't need to make this change!!!**

I'm just sharing with you the way I like to code because this way you can rid of `result`, do all the logic inside `reduce` and return it, and also avoid mutations... well, actually you have this mutation:
```js
result.push(...process(neighbor))
```
But in this case, I prefer it than
```js
(result, neighbor) => 
  [...result, ...process(neighbor)]
,
```
Because creating a new array in each interaction can be expensive and the mutation will happen just inside `reduce`.